app/src/components/BaseImageUpload.vue

```diff
@@ -1,16 +1,67 @@
 <template>
-  <div :class="width">
-    <label class="flex flex-col bg-white items-center block px-6 py-8 border border-grey-400 cursor-pointer">
-      <span>Choose image file to upload</span>
-      <div v-if="isValid && val" :id="`${id}-success`">Uploaded: {{ val.name }}</div>
-      <input type="file" @input="onInput" class="hidden" />
-    </label>
-    <div v-if="!isValid" class="bg-pink p-4 text-white" :id="`${id}-error`">{{ errorMsg }}</div>
+  <!-- Image Upload -->
+  <template v-if="!logoURI || !isValid">
+    <div class="block">
+      <label class="w-100 flex flex-col bg-white items-center px-6 py-8 border border-grey-400 cursor-pointer">
+        <input type="file" @input="onInput" class="hidden" />
+        <div class="flex p-12 gap-x-12">
+          <div
+            class="
+              group
+              flex
+              items-center
+              justify-center
+              h-24
+              border border-grey-400
+              p-12
+              cursor-pointer
+              hover:border-grey-500
+            "
+          >
+            <ImportIcon class="icon icon-small icon-primary" />
+          </div>
+          <div class="text-grey-400 flex items-center">
+            1920x1080px<br />
+            *.png
+          </div>
+        </div>
+      </label>
+    </div>
+
+    <div v-if="logoURI && !isValid" class="bg-pink p-4 text-white" :id="`${id}-error`">
+      {{ errorMsg }}
+    </div>
+  </template>
+
+  <template v-else-if="isUploading">
+    <span>Uploading to IPFS...</span>
+  </template>
```

Comment:
Seems this div needs to be before the `v-if="!logoURI || !isValid"`, since currently this is never shown if you upload a valid image—`isValid` starts as true by default, so if I upload a valid image, we still land in the above div while the image is uploading, then jump to the last div once uploaded

So it should be

```
<template v-if="isUploading">
...
<template v-else-if="!logoURI || !isValid">
...
<div v-else-if="logoURI" >
```

Also, is it possible to have a case where none of those three conditions are true and nothing ends up being displayed?

Reply:
Done